### PR TITLE
u/fjammes/DM-1974

### PR DESCRIPTION
**Pull request by andy-slac**
Manage ENCLOSED BY, NEWLINE characters in MySQL queries

### CODE REVIEW

File: admin/python/lsst/qserv/admin/dataLoader.py [hunk 1]
```diff
@@ -753,7 +761,7 @@ def _makeIndex(self, database, table):
         czarCursor.execute(q)
         for row in czarCursor.fetchall():
             if row[0] == idxCol:
-                idxColType = row[1]
+                IdxColType = row[1]
```
Comment (andy-slac) [inline admin/python/lsst/qserv/admin/dataLoader.py]: Something is wrong with capitalization here.

File: admin/python/lsst/qserv/admin/dataLoader.py [hunk 2]
```diff
@@ -801,7 +809,7 @@ def _makeIndexMultiNode(self, database, table, metaTable, idxCol):
                     break
                 czarCursor.executemany(q, seq)
 
-            wCursor.close()
+            Wcursor.close()
```
Comment (andy-slac) [inline admin/python/lsst/qserv/admin/dataLoader.py]: Same problem with capitalization.

File: admin/python/lsst/qserv/admin/dataLoader.py [hunk 3]
```diff
@@ -636,12 +636,20 @@ def _loadOneFile(self, conn, database, table, file, csvPrefix):
...
+        special_chars = map(repr, special_chars)
 
         self._log.info('load table %s from file %s', table, file)
-        q = "LOAD DATA LOCAL INFILE '%s' INTO TABLE %s.%s FIELDS TERMINATED BY '%s'" % \
-            (file, database, table, separator)
+        q = "LOAD DATA LOCAL INFILE '%s' INTO TABLE %s.%s" % (file,
+                                                              database,
+                                                              table)
+        q += (" FIELDS TERMINATED BY {0} ENCLOSED BY {1} ESCAPED BY {2}"
+              " LINES TERMINATED BY {3}").format(*special_chars)
```
Comment (andy-slac) [inline admin/python/lsst/qserv/admin/dataLoader.py]: The order of the arguments is not obvious here, what happens if someone re-oders `special_chars` list.

File: admin/python/lsst/qserv/admin/dataLoader.py [hunk 4]
```diff
@@ -636,12 +636,20 @@ def _loadOneFile(self, conn, database, table, file, csvPrefix):
...
 
-        # need to know field separator, default is the same as in partitioner.
-        separator = self.partOptions.get(csvPrefix + '.delimiter', '\t')
+        # need to know field delimiter, default is the same as in partitioner
+        special_chars = [self.partOptions.get(csvPrefix + '.delimiter', '\t'),
+                         self.partOptions.get(csvPrefix + '.enclose', '"'),
+                         self.partOptions.get(csvPrefix + '.escape', '\\'),
+                         self.partOptions.get(csvPrefix + '.newline', '\n')]
+
+        special_chars = map(repr, special_chars)
```
Comment (andy-slac) [inline admin/python/lsst/qserv/admin/dataLoader.py]: You also need to escape single quote character `'`, and `repr` will not do that.

**Comment by fritzm** [comment]
Hmm -- I tried checking out the parent of the merge commit for this PR, then force-pushing here to see if I could get github to "notice" the merge and close the PR.  But instead it scrubbed all the commits off the PR which I didn't expect...  perhaps I messed up my push here...

**Pull request closed without merging**